### PR TITLE
esp32c6: add wifi support

### PR DESCRIPTION
**Pull request by sylvioalves**
Bring ESP32-C6 Wi-Fi support.

### CODE REVIEW

**Review by wmrsouza — APPROVED** [review]

File: zephyr/esp32c6/CMakeLists.txt [hunk 1]
```diff
@@ -88,14 +88,36 @@ if(CONFIG_SOC_SERIES_ESP32C6)
     ../../components/heap/include
 
     ../../components/esp_mm/include
+
+    ../../components/esp_netif/include
+
+    ../../components/esp_coex/include
+    ../../components/esp_phy/include
+    ../../components/esp_phy/include/esp_private
+    ../../components/esp_phy/esp32c6/include
```
Comment (raffarost) [inline zephyr/esp32c6/CMakeLists.txt:97]: better using ${CONFIG_SOC_SERIES} here?

File: zephyr/esp32c6/CMakeLists.txt [hunk 2]
```diff
@@ -269,19 +293,264 @@ if(CONFIG_SOC_SERIES_ESP32C6)
...
   zephyr_sources_ifdef(
     CONFIG_UART_ESP32
     ../../components/hal/uart_hal.c
     ../../components/hal/uart_hal_iram.c
     )
-    
+
   zephyr_sources_ifdef(
     CONFIG_PWM_LED_ESP32
+    ../../components/soc/esp32c6/ledc_periph.c
```
Comment (raffarost) [inline zephyr/esp32c6/CMakeLists.txt:304]: here too?

File: zephyr/esp32c6/CMakeLists.txt [hunk 3]
```diff
@@ -269,19 +293,264 @@ if(CONFIG_SOC_SERIES_ESP32C6)
...
+
+  ## BT definitions
+  if (CONFIG_BT)
+    zephyr_sources(src/bt/esp_bt_adapter.c)
+    zephyr_compile_definitions(CONFIG_BT_ENABLED)
+
+    zephyr_link_libraries(
+      ## ble
+      btbb
+        -L${CMAKE_CURRENT_SOURCE_DIR}/../blobs/lib/esp32c6
```
Comment (raffarost) [inline zephyr/esp32c6/CMakeLists.txt:349]: maybe here too

**Review by raffarost — APPROVED** [review]

File: zephyr/esp32c6/src/wifi/esp_wifi_adapter.c [hunk 4]
```diff
@@ -0,0 +1,1076 @@
...
+	._sleep_retention_find_link_by_id = sleep_retention_find_link_by_id_wrapper,
+#endif
+	._coex_schm_process_restart = coex_schm_process_restart_wrapper,
+	._coex_schm_register_cb = coex_schm_register_cb_wrapper,
+    ._coex_schm_flexible_period_set = coex_schm_flexible_period_set_wrapper,
+    ._coex_schm_flexible_period_get = coex_schm_flexible_period_get_wrapper,
+	._magic = ESP_WIFI_OS_ADAPTER_MAGIC,
+};
+
+static void esp_wifi_free(void *mem)
```
Comment (marekmatej) [inline zephyr/esp32c6/src/wifi/esp_wifi_adapter.c:1073]: Whats the reason for burying this call down below? Could we move it to other heap calls?

File: components/esp_timer/private_include/esp_timer_impl.h [hunk 5]
```diff
@@ -19,7 +19,15 @@
 #include <stdint.h>
 #include "esp_err.h"
 
-typedef void (*intr_handler_t)(void *arg);
+#if defined(CONFIG_SOC_SERIES_ESP32C2) || \
+	defined(CONFIG_SOC_SERIES_ESP32C3) || \
+	defined(CONFIG_SOC_SERIES_ESP32C6)
+#include <zephyr/drivers/interrupt_controller/intc_esp32c3.h>
+#define ISR_HANDLER isr_handler_t
```
[the remark this answers is not in the document]
Comment (sylvioalves) [reply components/esp_timer/private_include/esp_timer_impl.h:26]: This is not as straightforward as it sounds. That will be part of a seconds round.

File: zephyr/esp32c6/src/wifi/esp_wifi_adapter.c (hunk 4 above)
Comment (sylvioalves) [reply zephyr/esp32c6/src/wifi/esp_wifi_adapter.c]: Sure, moved up to the same position as done in other adapters.

**Comment by marekmatej** [comment]
besides the nit comment LGTM

**Review by marekmatej — APPROVED** [review]

File: zephyr/esp32c6/src/wifi/esp_wifi_adapter.c [hunk 6]
```diff
@@ -0,0 +1,1077 @@
...
+	._regdma_link_set_write_wait_content = regdma_link_set_write_wait_content_wrapper,
+	._sleep_retention_find_link_by_id = sleep_retention_find_link_by_id_wrapper,
+#endif
+	._coex_schm_process_restart = coex_schm_process_restart_wrapper,
+	._coex_schm_register_cb = coex_schm_register_cb_wrapper,
+    ._coex_schm_flexible_period_set = coex_schm_flexible_period_set_wrapper,
+    ._coex_schm_flexible_period_get = coex_schm_flexible_period_get_wrapper,
+	._magic = ESP_WIFI_OS_ADAPTER_MAGIC,
+};
+
```
[the remark this answers is not in the document]
Comment (sylvioalves) [reply zephyr/esp32c6/src/wifi/esp_wifi_adapter.c]: Thanks!

**Review by marekmatej — APPROVED** [review]

**Review by wmrsouza — APPROVED** [review]

**Pull request merged**